pkg/api/pod/util.go

```diff
@@ -1231,6 +1231,10 @@ func hasInvalidLabelValueInAffinitySelector(spec *api.PodSpec) bool {
 }
 
 func MarkPodProposedForResize(oldPod, newPod *api.Pod) {
+	if len(newPod.Spec.Containers) != len(oldPod.Spec.Containers) {
```

Comment:
why not iterate through a merged list of containers with a container name existing on both newPod.Spec.Containers and oldPod.Spec.Containers?  (instead of ignoring the rest of execution when len is not equal)

Reply:
Since modifying the number of containers is not supported during pod updates, there is no need to perform any further operations if the lengths are unequal.

Reply:
Maybe add a comment to that effect?
```
// Update is invalid: ignore changes and let validation handle it
```

Reply:
Can you add this comment? Then I'll track down an API review for you :)

Reply:
> Maybe add a comment to that effect?

Sorry for missing this comment.